numpy/_globals.py

```diff
@@ -89,3 +92,21 @@ def __repr__(self):
 
 
 _NoValue = _NoValueType()
+
+class CopyMode(enum.IntEnum):
```

Comment:
Is there a reason why the new `CopyMode` needs to be an integer? Generally `IntEnum` exists to make it easier to preserve backwards compatibility with existing enum values represented as Python `int` objects. I don't think we have such concerns here, so I would just use `enum.Enum`.

Reply:
I will try if `enum.Enum` works. If it does then I would replace, `enum.IntEnum` with it. Thanks.

Reply:
I think, to allow, `True` and `False` to still work after introduction of this never copy mode, we should use, `enum.IntEnum`. I have included tests for verifying that, `True`/`False`, `np.CopyMode.ALWAYS`/`np.CopyMode.IF_NEEDED` and `np.True_`/`np.False_` behave similarly. From general perspective, if we are able to find a way which allows both the previous API and new extended API work, then I think we should go with it. Let me know if I am missing something here.

Reply:
We should definitely preserve backwards compatibility with boolean values passed into `copy`, but there's more than one way to do that.

My suggestion would be to normalize the `copy` argument with something like:
```python
if not isinstance(copy, CopyMode):
  copy = CopyMode.ALWAYS if copy else CopyMode.IF_NEEDED
```
(or the C equivalent)

Then all the internal logic in NumPy can just work with enum values, regardless of whether `CopyMode` is also an integer or can be cast to a boolean. This is generally a better idea than trying to make the full logic of how an argument is handled work with multiple argument types.

Reply:
I see. This would help in maintaining consistency as well. However, I am unable to figure how to convert a `PyObject` of type `CopyMode` to a C-enum in `PyArray_CopyConverter`([this line](https://github.com/numpy/numpy/pull/19173/files#diff-e3d34030ba0494a6677e227928e6114abda31ce5d4a48f23048114b481514cbbR177)). Also, how to do `isinstance(copy, CopyMode)` in C. There is a function `PyObject_IsInstance` ([here](https://docs.python.org/3/c-api/object.html#c.PyObject_IsInstance)). However, it accepts the class as `PyObject*`. So, is there any way to represent `CopyMode` as a `PyObject*` or even import it? Once that's clear, I will just extract the value of copy mode and map to corresponding C enum.

Reply:
I figured this out. Will be resolving this shortly.